axelor-account/src/main/java/com/axelor/apps/account/service/ReconcileGroupServiceImpl.java

```diff
@@ -245,6 +247,21 @@ && isBalanced(reconcileList)
     }
   }
 
+  @Override
+  public void letter(ReconcileGroup reconcileGroup) throws AxelorException {
+
+    List<MoveLine> moveLines =
+        moveLineRepository
+            .all()
+            .filter(
+                "(self.id IN (SELECT debitMoveLine FROM Reconcile debitReconcile WHERE debitReconcile.reconcileGroup.id = :reconcileGroupId AND debitReconcile.statusSelect = :confirmed))"
```

Comment:
moveLines should be get using moveLine.reconcileGroup relation. We can't use the Reconcile relation because the goal of this method is to create the Reconciles..